src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6390,12 +6404,17 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                             var boundType = BindNamespaceOrType(left, typeDiagnostics);
                             if (TypeSymbol.Equals(boundType.Type, leftType, TypeCompareKind.AllIgnoreOptions))
                             {
+                                Debug.Assert(!leftType.IsDynamic());
+                                Debug.Assert(IsPotentialColorColorReceiver(left, leftType));
```

Comment:
>Debug.Assert(IsPotentialColorColorReceiver(left, leftType)); [](http://example.com/codeflow?start=32&length=60)

What is the assert checking? (It looks like `IsPotentialColorColorReceiver()` is checking the same two conditions as the two `if` conditions above.)

Reply:
> What is the assert checking? (It looks like ```IsPotentialColorColorReceiver()``` is checking the same two conditions as the two if conditions above.)

It is verifying that we can rely on ```IsPotentialColorColorReceiver``` elsewhere to check for the same condition.